### PR TITLE
Improve properties verification of EncryptAlgorithm

### LINKED ISSUE #24747: Improve properties verification of EncryptAlgorithm
Hi community,

This issue is one of the tasks of #24746.

If not sure what to do, please refer to the example: #24742

## Aim
- [ ] MD5EncryptAlgorithm
- [ ] RC4EncryptAlgorithm
- [ ] AESEncryptAlgorithm
- [ ] SM4EncryptAlgorithm
- [ ] SM3EncryptAlgorithm
- [ ] CharDigestLikeEncryptAlgorithm

### PR DESCRIPTION
**Pull request by strawberry-crisis**
Fixes #24747.

Changes proposed in this pull request:
  - Improve properties verification of EncryptAlgorithm
---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/encrypt/RC4EncryptAlgorithm.java [hunk 1]
```diff
@@ -30,27 +31,26 @@
...
     private volatile int[] sBox = new int[SBOX_LENGTH];
-    
+
     @Override
     public void init(final Properties props) {
         reset();
         setKey(props.getProperty(RC4_KEY, "").getBytes(StandardCharsets.UTF_8));
     }
-    
+
```
Comment (Pace2Car) [inline features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/encrypt/RC4EncryptAlgorithm.java:50]: Please keep indentation consistent with the previous line, even empty lines

Comment (strawberry-crisis) [reply features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/encrypt/RC4EncryptAlgorithm.java:50]: OK, I'll fix it

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/encrypt/AESEncryptAlgorithm.java [hunk 2]
```diff
@@ -50,8 +50,10 @@ public void init(final Properties props) {
     }
     
     private byte[] createSecretKey(final Properties props) {
-        ShardingSpherePreconditions.checkState(props.containsKey(AES_KEY), () -> new EncryptAlgorithmInitializationException("AES", String.format("%s can not be null", AES_KEY)));
-        return Arrays.copyOf(DigestUtils.sha1(props.getProperty(AES_KEY)), 16);
+        String aesKey = props.getProperty(AES_KEY);
+        ShardingSpherePreconditions.checkState(null != aesKey && !aesKey.isEmpty(),
```
Comment (Pace2Car) [inline features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/encrypt/AESEncryptAlgorithm.java:54]: Here you can use guava `!Strings.isNullOrEmpty()` instead

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/encrypt/RC4EncryptAlgorithm.java [hunk 3]
```diff
@@ -30,27 +31,26 @@
...
     private void setKey(final byte[] key) {
-        if (!(key.length >= KEY_MIN_LENGTH && key.length < SBOX_LENGTH)) {
-            throw new EncryptAlgorithmInitializationException("RC4", "Key length has to be between " + KEY_MIN_LENGTH + " and " + (SBOX_LENGTH - 1));
-        }
+        ShardingSpherePreconditions.checkState(key.length >= KEY_MIN_LENGTH && key.length < SBOX_LENGTH,
```
Comment (Pace2Car) [inline features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/encrypt/RC4EncryptAlgorithm.java:52]: Keep constants is in front of logical symbols

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/encrypt/AESEncryptAlgorithm.java [hunk 4]
```diff
@@ -50,8 +50,10 @@ public void init(final Properties props) {
...
+                () -> new EncryptAlgorithmInitializationException(getType(), String.format("%s can not be null or empty", AES_KEY)));
```
Comment (Pace2Car) [inline features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/encrypt/AESEncryptAlgorithm.java:55]: Please add a test case where AES_KEY is empty

**Review by Pace2Car — APPROVED** [review]

**Review by RaigorJiang — APPROVED** [review]
